test/concurrent/ThreadPoolTest.ooc

```diff
@@ -27,6 +27,11 @@ ThreadPoolTest: class extends Fixture {
 			expect(promise3 wait())
 			expect(promise4 wait(), is false)
 			expect(promise5 wait(), is true)
+			expect(promise1 _state as Int, is equal to(_PromiseState Finished as Int))
+			expect(promise2 _state as Int, is equal to(_PromiseState Finished as Int))
+			expect(promise3 _state as Int, is equal to(_PromiseState Finished as Int))
+			expect(promise4 _state as Int, is equal to(_PromiseState Cancelled as Int))
+			expect(promise5 _state as Int, is equal to(_PromiseState Finished as Int))
```

Comment:
Sounds a lot better as moving ```_state``` to ```_ThreadPromise``` and ```ThreadFuture<T>``` hides it from ```_Threadpool```. Although I'm not sure we need to have a ```getState()``` method since no code (currently) make use of the state outside the class itself (neglecting my newly invented test case).